src/server.jsx

```diff
@@ -166,6 +166,13 @@ function setupServer(req, res, next) {
     config.settings.publicURL = res.locals.detectedHost;
   }
 
+  if (!process.env.RAZZLE_INTERNAL_API_PATH && req.headers['x-internal-host']) {
+    res.locals.detectedInternalHost = `${
+      req.headers['x-forwarded-proto'] || req.protocol
+    }://${req.headers['x-internal-host']}`;
+    config.settings.internalApiPath = res.locals.detectedInternalHost;
```

Comment:
The thing that we're unsure of (at least I am) and needs testing is this line. 

Basically, I think it would be a good idea that all our URL helper functions stop depending on config.settings.apiPath and config.settings.internalApiPath and always use information either from the browser window or server HOST. CC @sneridagh

Reply:
Yeah we should talk about it during the conference.
I added it to my todo list.